onnxruntime/python/tools/transformers/fusion_bart_attention.py

```diff
@@ -149,12 +240,55 @@ def fuse(self, normalize_node, input_name_to_nodes, output_name_to_node):
             ["Reshape"],
             [1],
         )
+        v_nodes_with_past_cross_attn_openai = self.model.match_parent_path(
+            matmul_qkv,
+            ["Transpose", "Reshape", "Reshape", "Transpose"],
+            [1, 0, 0, 0],
+        )
         past_v, present_v = "", ""
         reshape_v_2, add_v = None, None
         if v_nodes is not None:
             (reshape_v_2, transpose_v, reshape_v_1, add_v, matmul_v) = v_nodes
             # For initial pass through encoder-decoder_with_past to get starting past values (beam search)
             present_v = transpose_v.output[0]
+        elif v_nodes_openai is not None:
+            v_nodes = v_nodes_openai
+            (transpose_v, reshape_v_1, add_v, matmul_v) = v_nodes
+            # For initial pass through encoder-decoder_with_past to get starting past values (beam search)
+
+            # Find the child path to access the correct present_v values
+            # Openai impl provides present/past v values in 3D format
+            # whereas ort MultiHeadAttention expects v values in 4D, hence the
+            # additional Reshape and Transpose nodes are added
+            # For encoder attention types
+            # Add -> Reshape -> Transpose -> Present_V
+            reshape_path = self.model.match_child_path(
```

Comment:
Let's pass the indices to the `match_child_path` method to follow what `match_parent_path` does.

Reply:
The passing of the child indices might be unreliable in this case, for eg the pattern looks like
```
Add -> Reshape -> Transpose -> present_val (left path [0, 0])
        |-> Reshape -> Transpose -> MatMul (right path [1, 0])
```
and in some cases it looks like
```
Add -> Reshape -> Transpose -> MatMul (left path [0, 0])
        |-> Reshape -> Transpose -> present_val (right path [1, 0])
```
hence the value of the indices cannot always be fixed.
The exclude parameter should be sufficient to determine the correct child path to be chosen in this case